### PR TITLE
Batch Operation - start worker 

### PR DESCRIPTION
**Pull request by longquanzheng**

### CODE REVIEW

File: common/constants.go [hunk 1]
```diff
@@ -86,12 +86,16 @@ const (
 )
 
 const (
-	// SystemDomainName is domain name for all cadence system workflows
+	// SystemGlobalDomainName is global domain name for cadence system workflows running globally
+	SystemGlobalDomainName = "cadence-system-global"
+	// SystemDomainName is domain name for cadence system workflows running in local cluster
 	SystemDomainName = "cadence-system"
```
Comment (wxing1292) [inline common/constants.go]: plz also change this to local

File: common/service/dynamicconfig/constants.go [hunk 2]
```diff
@@ -197,6 +197,7 @@ var keys = map[Key]string{
 	WorkerTimeLimitPerArchivalIteration:             "worker.TimeLimitPerArchivalIteration",
 	WorkerThrottledLogRPS:                           "worker.throttledLogRPS",
 	ScannerPersistenceMaxQPS:                        "worker.scannerPersistenceMaxQPS",
+	EnableBatcher:                                   "worker.enableBatcher",
```
Comment (wxing1292) [inline common/service/dynamicconfig/constants.go]: plz move this to section `// system settings`, around line 53

File: service/worker/batcher/batcher.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+// THE SOFTWARE.
+
+package batcher
+
+import (
+	"context"
+	"time"
+
+	"github.com/uber/cadence/common/service/dynamicconfig"
+
```
Comment (wxing1292) [inline service/worker/batcher/batcher.go]: empty line

File: service/worker/batcher/batcher.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+	if err == nil {
+		s.context.logger.Info("Global system domain already exists", tag.WorkflowDomainName(common.SystemGlobalDomainName))
+		return nil
+	}
+
+	ctx, cancel = context.WithTimeout(context.Background(), rpcTimeout)
+	err = s.context.svcClient.RegisterDomain(ctx, s.getDomainCreationRequest())
+	cancel()
+	if err != nil {
+		s.context.logger.Error("Error creating global system domain", tag.Error(err))
```
Comment (wxing1292) [inline service/worker/batcher/batcher.go]: maybe should check if domain not active error is returned here

depends on how you want to deal with global domain creation, there can be several things
1. if seeing domain not active here, loop + backoff + retry on describe domain, waiting on worker to create the domain on active side
2. crash.

File: service/worker/batcher/batcher.go [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+		EmitMetric:                             common.BoolPtr(true),
+		SecurityToken:                          common.StringPtr(s.context.cfg.AdminOperationToken()),
+		IsGlobalDomain:                         common.BoolPtr(false),
+	}
+
+	if s.context.cfg.ClusterMetadata.IsGlobalDomainEnabled() {
+		req.IsGlobalDomain = common.BoolPtr(true)
+		req.ActiveClusterName = common.StringPtr(s.context.cfg.ClusterMetadata.GetMasterClusterName())
+		var clusters []*shared.ClusterReplicationConfiguration
+		for name := range s.context.cfg.ClusterMetadata.GetAllClusterInfo() {
```
Comment (wxing1292) [inline service/worker/batcher/batcher.go]: plz also check inside the cluster info whether the cluster is enabled

File: common/service/dynamicconfig/constants.go (hunk 2 above)
Comment (longquanzheng) [reply common/service/dynamicconfig/constants.go]: Sure

File: service/worker/batcher/batcher.go (hunk 4 above)
Comment (longquanzheng) [reply service/worker/batcher/batcher.go]: Sure

File: service/worker/batcher/batcher.go (hunk 5 above)
Comment (longquanzheng) [reply service/worker/batcher/batcher.go]: Sure

File: common/constants.go (hunk 1 above)
Comment (longquanzheng) [reply common/constants.go]: that will be related to the existing domain we created in production. Let me create an issue and talk to the team.

Comment (longquanzheng) [reply common/constants.go]: https://github.com/uber/cadence/issues/2000

File: service/worker/batcher/batcher.go (hunk 4 above)
Comment (longquanzheng) [reply service/worker/batcher/batcher.go]: the error we are getting is:
```
errNotMasterCluster                = &gen.BadRequestError{Message: "Cluster is not master cluster, cannot do domain registration or domain update."}
```
do you want to check on badRequest and the message inside?

**Pull request merged**